### DISCUSSION TITLE
Poor performance concerns

### DISCUSSION THREAD

**Discussion by JoeyFenny** [rnmapbox/maps#1303] [Q&A] [answered]
Is performance atrocious for anyone else? Simple things like pinch to zoom and dragging on iOS simulator lag incredibly bad, compared to my experience using react-native-maps or native

**Comment by systemlevel** [comment]
Yes. But I don’t know if the root cause of that is the react native layer or not.

**Reply by JoeyFenny** [reply]
Do we know of any apps on the iOS or Android app stores that are using this in production? I want to see how it performs on their production app.

**Reply by systemlevel** [reply]
Yes, we use it in production.  The simulator is the only thing that is slow.  The actual devices are fine and work well.

**Comment by ferdicus** [comment, marked as answer]
Simply check out the repo and build the example app on your physical device. 
No need to play around with any third party apps when you have an app in the project.

Then you can decide if the performance is good enough for your needs

**Comment by jaltin** [comment]
This has already been answered but I wanted to add a comment in case someone else concerned with performance stumbles on this post and are considering alternatives due to performance worries.

We have used mapbox react native for several years now and are really happy with it.

The performance on a real device is in my experience REALLY GREAT compared to the native alternatives. This becomes even more obvious once you start adding heavy/many map features on top of the background map.

**Comment by Cmoen11** [comment]
We've been using this for the past year. And the only thing we've noticed is that the phone gets hot while used. Performance is good.
